### ISSUE TITLE
Automatic link to typedef no longer generated if typedef declared twice

### ISSUE THREAD

**Issue by abbeybreeding** [doxygen/doxygen#11010]
**Describe the bug**
We had a typedef declared twice, one with documentation and the other without. If the one without is first in the path, the link to the documentation for the typedef will not be generated.

**Screenshots**
1.8.9.1
![image](https://github.com/user-attachments/assets/05a7f387-03cb-4c1c-a15c-0bafe0df8d15)
1.11.0
![image](https://github.com/user-attachments/assets/586197d9-f53d-4e80-bec6-c4fe00474b34)

**To Reproduce**
[testcase.zip](https://github.com/user-attachments/files/16289294/testcase.zip)

**Expected behavior**
While our code is incorrect, and we have addressed that, it would still seem that the link to the typedef documentation would still be generated.

**Version**
This seemed to "work" as we would expect in 1.8.9.1 (see screenshot). It no longer "works" in 1.11.0 or the trunk.

**Additional context**
Various things, such as changing the path, adding a `\typedef` tag, sometimes adding a `\ref` tag with work around the issue.

**Comment by albert-github** [comment]
Did a quick "bisect" and it looks like that the commit (b290399fab92df45376103d7c094f053f24495fa ):
```
Commit: b290399fab92df45376103d7c094f053f24495fa [b290399]
Date: Tuesday, May 24, 2022 8:12:07 AM
Refactoring: reimplement getDefs using symbol resolver
```

Causes the problem.

@abbeybreeding when specifying trunk / master always specify the full version as this is a moving target, so `doxygen -v`.

I ran with the current master (1.12.0 (0b557b30eee328189629b2d3cc89f4774f3ad5ed)) and the problem is still present.

Might be related to:
- #10180 
- #9379

the fix for 
- #10994 

didn't solve this issue.

**Comment by abbeybreeding** [comment]
@albert-github  I'm so sorry about not adding the `doxygen -v` to the trunk version, I will make sure to do so in the future!  :-)

**Comment by doxygen** [comment]
@abbeybreeding Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by abbeybreeding** [comment]
@doxygen Hi Dmitri.  This does fix the example in the test case for us. Thanks to you and the team for fixing this.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
